### PR TITLE
Convert errors to coded exceptions in `@solana/keys`

### PR DESCRIPTION
**Pull request by steveluscher**
Addresses #2118.

### CODE REVIEW

**Review by buffalojoec — APPROVED** [review]
keys

File: packages/errors/src/codes.ts [hunk 1]
```diff
@@ -51,6 +51,10 @@ export const SOLANA_ERROR__INVARIANT_VIOLATION_WEBSOCKET_MESSAGE_ITERATOR_MUST_N
     3507001 as const;
 export const SOLANA_ERROR__INVARIANT_VIOLATION_CACHED_ABORTABLE_ITERABLE_CACHE_ENTRY_MISSING = 3507002 as const;
 export const SOLANA_ERROR__INVARIANT_VIOLATION_SWITCH_MUST_BE_EXHAUSTIVE = 3507003 as const;
+// Reserve key-related error codes in the range [3704000-3704999]
+export const SOLANA_ERROR__KEYS_PRIVATE_KEY_BYTE_LENGTH_OUT_OF_RANGE = 3704000 as const;
+export const SOLANA_ERROR__KEYS_SIGNATURE_BYTE_LENGTH_OUT_OF_RANGE = 3704001 as const;
+export const SOLANA_ERROR__KEYS_SIGNATURE_STRING_LENGTH_OUT_OF_RANGE = 3704002 as const;
```
Comment (lorisleiva) [inline packages/errors/src/codes.ts:57]: I think only `SIGNATURE_STRING_LENGTH_OUT_OF_RANGE` is a valid "out of range" errors. It feels weird to me to get `SIGNATURE_BYTE_LENGTH_OUT_OF_RANGE` for signature bytes. In fact, just reading this PR, I let myself thinking "What? I didn't know our signatures could be anything else than 64 bytes long". So I think we should make the following change:

```suggestion
export const SOLANA_ERROR__KEYS_INVALID_PRIVATE_KEY_BYTE_LENGTH = 3704000 as const;
export const SOLANA_ERROR__KEYS_INVALID_SIGNATURE_BYTE_LENGTH = 3704001 as const;
export const SOLANA_ERROR__KEYS_SIGNATURE_STRING_LENGTH_OUT_OF_RANGE = 3704002 as const;
```

**Review by lorisleiva — APPROVED** [review]
Aside from one naming comment, I'm happy with this!

**Comment by steveluscher** [comment]
### Merge activity

* **Mar 4, 3:29 PM PST**: @steveluscher started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/2245).

**Pull request merged**